### PR TITLE
Implementation of meanshift

### PR DESCRIPTION
**Pull request by bdeonovic**
This is an implementation of meanshift algorithm (for gaussian kernels)

Chen, Y. (1995). Mean Shift, Mode Seeking, and Clustering. IEEE Transactions on Pattern Analysis
and Machine Intelligence, 17, 790-799.

based off Jochen Einbeck and Ludger Evers implementation in their [LPCM R package](http://cran.r-project.org/web/packages/LPCM/index.html). I didn't implement the bandwidth selection.

### CODE REVIEW

**Comment by tlnagy** [comment]
@johnmyleswhite @KristofferC Have you guys had a chance to look at this? It's been sitting in limbo for almost a year. Is there any reason this hasn't been merged?

File: src/meanshift.jl [hunk 1]
```diff
@@ -0,0 +1,153 @@
+
+type MeanShiftResult{T <: FloatingPoint} <: ClusteringResult
```
Comment (tlnagy) [inline src/meanshift.jl]: `FloatingPoint` is soon to be deprecated in favor of `AbstractFloat`. See https://github.com/JuliaLang/julia/pull/12162

**Comment by anneaux** [comment]
Why is this closed? It's not actually merged into the master branch, is it?

**Comment by bdeonovic** [comment]
I think at the time I closed it because I did not have any more need for it and I did not want to update the code to get it pulled in (since there didn't seem to be any interest)

**Pull request closed without merging**